packages/ui/src/core/textarea/textarea.component.tsx

```diff
@@ -89,16 +97,33 @@ const Textarea = intrinsicComponent<TextareaProps, HTMLTextAreaElement>(
           disabled={disabled}
           style={{ ...overflowStyles }}
         />
-        {readOnly ? (
-          <Styled.CopyIcon
-            showCopyIcon={isHovering && value.length > 0}
-            size={size}
-            onClick={() => handleCopyIcon(value, setShowCopyMessage)}
-          >
-            <CopyOutline size={getIconSize(size)} />
-          </Styled.CopyIcon>
-        ) : undefined}
-        {showCopyMessage && renderCopyText()}
+
+        {(showActionButton || showClearButton || showCopyIcon) && (
+          <Styled.ActionsButtonsWrapper>
+            {showActionButton && (
+              <Button color="link-primary" size="sm" onClick={onClickActionButton}>
+                {actionButtonLabel}
+              </Button>
+            )}
+
+            {showClearButton && (
+              <Button color="link-secondary" size="sm" onClick={onClear}>
```

Comment:
onClick here should have a default action for clearing the value and then you can add an extra event like onClear for the user to add an extra action after clearing